rel/expr.go

```diff
@@ -21,6 +21,10 @@ type ContextErr struct {
 	scope  Scope
 }
 
+func NewContextErr(err error, source parser.Scanner, scope Scope) ContextErr {
```

Comment:
ContextErr seems to have become a much more fundamental type. Could you add a brief doc explaining its purpose? It seems strange that you have to pass an array of "errors" to create a new shell instance, for example.

Reply:
added more comments on `ContextErr`. Passing an array does look strange, it's supposed to represent the frames, is it better to pass the actual error?

Reply:
I don't know enough about how it's used. Generally a conceptually simpler API is better, so a struct wrapping an array rather than a raw array is less to think about, but this is fine for now IMO.